### ISSUE TITLE
don't buffer when separator = ""

### ISSUE THREAD

**Issue by irickt** [lucagrulla/node-tail#18]
Here's a suggestion:

``` coffeescript
        stream.on 'data', (data) =>
          @buffer += data
          if @separator is ''
            # just send the whole buffer
            # (without this condition it buffers the last character)
            @emit("line", @buffer) # "data" event to be more precise
            @buffer = ''
          else
            # buffer anything after the last separator
            # (and don't send the separators)
            parts = @buffer.split(@separator)
            @buffer = parts.pop()
            @emit("line", chunk) for chunk in parts
```

**Comment by dceejay** [comment]
Did you ever get round to thinking about this ?
It would also be useful for tailing binary files which by their nature could contain any character - so you ideally want to just spit out the whole new chunk that arrives... including the last character,  but at least if it sent every character one at a time (as now) it would be ok.

**Comment by lucagrulla** [comment]
Hi @dceejay,

No I have not solved this.

**Comment by lucagrulla** [comment]
Fixed with release 1.4.0. To avoid buffering just pass null as parameter in the constructor.
